lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -423,17 +425,21 @@ def add_network_token_data(post_data, payment_method, options)
         post_data
       end
 
-      def add_network_token_cryptogram_and_eci(post, payment_method)
-        return unless adding_network_token_card_data?(payment_method)
+      def add_network_token_cryptogram_and_eci(post, payment_method, options)
+        return unless !options[:wallet_type] || adding_network_token_card_data?(payment_method, options)
 
         post[:payment_method_options] ||= {}
         post[:payment_method_options][:card] ||= {}
         post[:payment_method_options][:card][:network_token] ||= {}
-        post[:payment_method_options][:card][:network_token][:cryptogram] = payment_method.payment_cryptogram if payment_method.payment_cryptogram
-        post[:payment_method_options][:card][:network_token][:electronic_commerce_indicator] = payment_method.eci if payment_method.eci
+        post[:payment_method_options][:card][:network_token] = {
```

Comment:
Also if `post[:payment_method_options][:card][:network_token]` already exist we would have to use `merge!` because here we are resetting the object.

Reply:
So you could do something like

```
post[:payment_method_options][:card][:network_token].merge!({
  cryptogram: payment_method.respond_to?(:payment_cryptogram) ? payment_method.payment_cryptogram : options[:cryptogram],
          electronic_commerce_indicator: format_eci(payment_method, options)
 }.compact)
```

Reply:
Done!